### PR TITLE
Switch to newer json-patch; remove dead code

### PR DESCRIPTION
**Pull request by swenson**
Switch from `github.com/mattbaird/jsonpatch` (unmaintained) to a new version of `github.com/evanphx/json-patch`.

I also noticed that the `agent.Patches` object is unused, as is the `patches` parameter of `agent.New()`, so I removed those, which simplifies some of the code.

(This is in preparation for some more work involving JSON patches, so I wanted to clean up existing usage of it.)

### CODE REVIEW

File: agent-inject/handler_test.go [hunk 1]
```diff
@@ -601,13 +401,16 @@ func TestHandlerHandle(t *testing.T) {
...
 			if len(resp.Patch) > 0 {
 				req.NoError(json.Unmarshal(resp.Patch, &actual))
 				for i := range actual {
-					actual[i].Value = nil
+					delete(actual[i], "value")
 				}
 			}
+			for i := range tt.Patches {
+				delete(tt.Patches[i], "value")
+			}
```
Comment (tvoran) [inline agent-inject/handler_test.go:413]: Was this extra `delete` loop for the expected patches necessary with the new jsonpatch, or just being prudent?

File: agent-inject/agent/agent.go [hunk 2]
```diff
@@ -524,49 +518,47 @@ func ShouldInject(pod *corev1.Pod) (bool, error) {
 
 // Patch creates the necessary pod patches to inject the Vault Agent
 // containers.
-func (a *Agent) Patch() ([]byte, error) {
-	var patches []byte
-
+func (a *Agent) Patch() (patches jsonpatch.Patch, patchesBytes []byte, err error) {
```
Comment (tvoran) [inline agent-inject/agent/agent.go:521]: Is this returning `patches jsonpatch.Patch` just for use in tests? I'm not seeing that parameter used in the actual code. I'm wondering if it needs to be returned, and patchesBytes could just be unmarshal'ed in the test.

**Review by tvoran — COMMENTED** [review]
This looks pretty great; love the internal json patch helpers. Just a couple style comments so far.

File: agent-inject/agent/agent.go [hunk 3]
```diff
@@ -584,9 +576,10 @@ func (a *Agent) Patch() ([]byte, error) {
 
 	// Init Container
 	if a.PrePopulate {
-		container, err := a.ContainerInitSidecar()
+		var container corev1.Container
+		container, err = a.ContainerInitSidecar()
 		if err != nil {
-			return patches, err
+			return
```
[the remark this answers is not in the document]
Comment (swenson) [reply agent-inject/agent/agent.go:582]: Yeah, in general I agree. I'll see if I can rework this.

File: agent-inject/handler_test.go (hunk 1 above)
Comment (swenson) [reply agent-inject/handler_test.go:413]: It's necessary now to remove the value field from the test cases. I tried passing in `nil`, but it still generates some bytes (JSON `null`) I think, so the two won't be equal below.

**Review by tomhjp — APPROVED** [review]
LGTM!

**Review by tvoran — APPROVED** [review]

**Comment by swenson** [comment]
Thanks!

**Pull request merged**